pandapower/results_branch.py

```diff
@@ -578,6 +581,67 @@ def _get_impedance_results(net, ppc, i_ft, suffix=None):
     res_impedance_df["i_to_ka"].values[:] = i_to_ka
 
 
+def _get_tcsc_results(net, ppc, suffix=None):
+    if len(net.tcsc) == 0:
+        return
+
+    ac = net["_options"]["ac"]
+
+    baseMVA = ppc["baseMVA"]
+    bus_lookup = net["_pd2ppc_lookups"]["bus"]
+    f = 0
+    t = len(net.tcsc)
+
+    f_bus = bus_lookup[net.tcsc["from_bus"].values]
+    t_bus = bus_lookup[net.tcsc["to_bus"].values]
+    baseZ = ppc["bus"][f_bus, BASE_KV] ** 2 / baseMVA
+
+    if ac:
+        p_from_mw = ppc["tcsc"][f:t, TCSC_PF].real
+        p_to_mw = ppc["tcsc"][f:t, TCSC_PT].real
+        q_from_mvar = ppc["tcsc"][f:t, TCSC_QF].real
+        q_to_mvar = ppc["tcsc"][f:t, TCSC_QT].real
+        i_from_ka = ppc["tcsc"][f:t, TCSC_IF].real
+        i_to_ka = ppc["tcsc"][f:t, TCSC_IT].real
+        # losses will be 0 because tcsc represents an ideal device but we can add resistance easily in the future
+        pl_mw = p_from_mw + p_to_mw
+        ql_mvar = q_from_mvar + q_to_mvar
+    else:
+        raise NotImplementedError("TCSC not implemented for algorithm != 'ac' - writing zeros to tcsc results. "
+                                  "Results for other elements are as if TCSC elements were open switches.")
+        zeros = np.zeros(t)
+        # this looks like a pyramid
+        p_from_mw = zeros
```

Comment:
A note just to force a check by @rbolgaryn : Might this reusage of the same variable `zeros` lead to undesired behavior?